sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -286,44 +322,49 @@ class RocksDB(
    */
   def commit(): Long = {
     val newVersion = loadedVersion + 1
-    val checkpointDir = createTempDir("checkpoint")
-    var rocksDBBackgroundThreadPaused = false
     try {
-      // Make sure the directory does not exist. Native RocksDB fails if the directory to
-      // checkpoint exists.
-      Utils.deleteRecursively(checkpointDir)
 
       logInfo(s"Flushing updates for $newVersion")
-      val flushTimeMs = timeTakenMs { db.flush(flushOptions) }
-
-      val compactTimeMs = if (conf.compactOnCommit) {
```

Comment:
We still need to retain this option for the non changelog checkpointing case right ? Otherwise, where would the `compactOnCommit` option be used ?

Reply:
I'm not sure we can just get rid of this directly though. Maybe we can start the deprecation process, but we might need to keep supporting this for a while ?

Reply:
Added back.